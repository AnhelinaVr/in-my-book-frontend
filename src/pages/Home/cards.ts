import Anhelina from './images/heroes/anhelina_vrubleuskaya.jpg';
import CardImage1 from './images/about/image-1.svg';
import CardImage2 from './images/about/image-2.svg';
import CardImage3 from './images/about/image-3.svg';
import { Cards } from './components/Cards';
import { ComponentProps } from 'react';
import Konstantin from './images/heroes/konstantin_shumak.jpg';
import Olga from './images/heroes/olga_polyanskaya.jpg';
import { Paths } from '../../constants';

export const aboutCards: ComponentProps<typeof Cards>['cards'] = [
  {
    title: 'Join',
    image: CardImage1,
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.',
    button: { text: 'Lorem Ipsum', path: Paths.home },
  },
  {
    title: 'Communicate',
    image: CardImage2,
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.',
    button: { text: 'Lorem Ipsum', path: Paths.home },
  },
  {
    title: 'Have fun!',
    image: CardImage3,
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.',
    button: { text: 'Lorem Ipsum', path: Paths.home },
  },
];

export const heroesCards: ComponentProps<typeof Cards>['cards'] = [
  {
    subtitle: 'Anhelina Vrubleuskaya',
    image: Anhelina,
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.',
  },
  {
    subtitle: 'Olga Polyanskaya',
    image: Olga,
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.',
  },
  {
    subtitle: 'Konstantin Shumak',
    image: Konstantin,
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.',
  },
];
