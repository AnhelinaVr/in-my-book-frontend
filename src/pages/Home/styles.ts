import { SxProps, Theme } from '@mui/material';
import HeroesBackground from './images/heroes/background.png';

export const getStyles = (theme: Theme): Record<string, SxProps> => ({
  mainContainer: {
    position: 'relative',
    height: '100vh',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    paddingLeft: theme.spacing(14),
    '& p, h2, h5': {
      position: 'relative',
      maxWidth: theme.spacing(100),
    },
    background: `linear-gradient(
      ${theme.palette.background.light} 0%, 
      ${theme.palette.background.light} 90%, 
      ${theme.palette.background.lighter}
      )`,
    [theme.breakpoints.down('desktop')]: {
      justifyContent: 'start',
      alignItems: 'center',
      textAlign: 'center',
    },
    [theme.breakpoints.down('tablet')]: {
      fontSize: 14,
      padding: theme.spacing(0, 8),
    },
  },
  mainContent: {
    marginTop: theme.spacing(31),
  },
  mainTitle: {
    fontSize: 72,
    margin: 0,
    [theme.breakpoints.down('tablet')]: {
      fontSize: 60,
    },
  },
  mainDescription: {
    marginTop: theme.spacing(10),
    background: `rgba(241, 224, 208, 0.65)`,
    borderRadius: theme.spacing(2),
    padding: theme.spacing(2),
  },
  mainImage: {
    position: 'absolute',
    maxWidth: theme.spacing(196.5),
    bottom: 0,
    width: '100%',
    right: 0,
  },
  title: {
    textAlign: 'center',
    marginBottom: theme.spacing(9),
    fontWeight: theme.typography.fontWeightBold,
    textTransform: 'uppercase',
    color: theme.palette.text.main,
    [theme.breakpoints.down('tablet')]: {
      fontSize: 32,
    },
  },
  about: {
    marginTop: theme.spacing(20),
    display: 'block',
    position: 'relative',
    [theme.breakpoints.down('desktop')]: {
      marginTop: theme.spacing(20),
    },
    [theme.breakpoints.down('tablet')]: {
      marginTop: theme.spacing(13),
    },
  },
  aboutBackground: {
    width: '100%',
    position: 'absolute',
    height: theme.spacing(62),
    zIndex: '-1',
    backgroundColor: theme.palette.background.dark,
    [theme.breakpoints.down('tablet')]: {
      display: 'none',
    },
  },
  heroes: {
    margin: theme.spacing(20, 0),
    [theme.breakpoints.down('desktop')]: {
      margin: theme.spacing(20, 0),
    },
    [theme.breakpoints.down('tablet')]: {
      margin: theme.spacing(13, 0),
    },
  },
  heroesContainer: {
    background: theme.palette.background.main,
    backgroundImage: `url(${HeroesBackground})`,
    padding: theme.spacing(1),
  },
});
