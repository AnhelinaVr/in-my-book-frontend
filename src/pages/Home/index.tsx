import { Box, Typography, useTheme } from '@mui/material';
import { aboutCards, heroesCards } from './cards';
import { CSSProperties } from 'react';
import { Cards } from './components/Cards';
import MainBackground from './images/main.jpg';
import { getStyles } from './styles';

const Home = () => {
  const theme = useTheme();
  const styles = getStyles(theme);

  return (
    <>
      <Box sx={styles.main}>
        <img
          style={styles.mainImage as CSSProperties}
          src={MainBackground}
          alt="background"
        />
        <Typography variant="h2" sx={{ color: 'secondary.main', fontSize: 72 }}>
          BOOKS
        </Typography>
        <Typography variant="h2" sx={{ color: 'primary.main', fontSize: 72 }}>
          club
        </Typography>
        <h3 style={{ marginTop: theme.spacing(11.5) }}>
          This application allows you to unite people with a common interest -
          reading books. The functionality allows users to share their opinions
          by writing book reviews, allows you to create rooms for joint
          discussion of books with the possibility of creating a video chat. In
          addition, the functionality of the application allows you to make
          personal notes about the book you read
        </h3>
      </Box>
      <Box sx={styles.about}>
        <Typography variant="h2" sx={styles.title}>
          About Us
        </Typography>
        <Box sx={styles.aboutBackground}></Box>
        <Cards cards={aboutCards} />
      </Box>
      <Box sx={styles.heroes}>
        <Typography variant="h2" sx={styles.title}>
          Our Heroes
        </Typography>
        <Box sx={styles.heroesContainer}>
          <Cards cards={heroesCards} />
        </Box>
      </Box>
    </>
  );
};

export default Home;
