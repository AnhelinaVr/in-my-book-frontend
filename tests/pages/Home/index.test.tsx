import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Home from '../../../pages/Home';

it('renders welcome message', () => {
  render(<Home />);

  expect(screen.getByText('Home')).toBeInTheDocument();
});
