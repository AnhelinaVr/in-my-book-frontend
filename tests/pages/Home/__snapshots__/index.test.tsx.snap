// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1fvd2r1"
  >
    <img
      alt="background"
      src="main.png"
      style="position: absolute; max-width: 1572px; bottom: 0px; width: 100%; right: 0px;"
    />
    <div
      class="MuiBox-root css-t8vfqg"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-1rx97d8-MuiTypography-root"
      >
        BOOKS
      </h2>
      <h2
        class="MuiTypography-root MuiTypography-h2 css-y94pyy-MuiTypography-root"
      >
        club
      </h2>
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1ncwoai-MuiTypography-root"
      >
        This application allows you to unite people with a common interest - reading books. The functionality allows users to share their opinions by writing book reviews, allows you to create rooms for joint discussion of books with the possibility of creating a video chat. In addition, the functionality of the application allows you to make personal notes about the book you read
      </h5>
    </div>
  </div>
  <div
    class="MuiBox-root css-10itben"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 css-131agbk-MuiTypography-root"
    >
      About Us
    </h2>
    <div
      class="MuiBox-root css-f3scve"
    />
    <div
      class="MuiBox-root css-3hhxqo"
    >
      <div
        class="MuiBox-root css-1wkb7z3"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-oxfbg2-MuiTypography-root"
        >
          Join
        </h3>
        <img
          alt="card"
          src="function SvgImage1(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgImage1.name})
  );
}"
          style="max-width: 400px; flex: 1; margin-bottom: 24px; object-fit: cover;"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 css-1no9wdb-MuiTypography-root"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1ckkjjs-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Lorem Ipsum
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-1wkb7z3"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-oxfbg2-MuiTypography-root"
        >
          Communicate
        </h3>
        <img
          alt="card"
          src="function SvgImage2(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgImage2.name})
  );
}"
          style="max-width: 400px; flex: 1; margin-bottom: 24px; object-fit: cover;"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 css-1no9wdb-MuiTypography-root"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1ckkjjs-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Lorem Ipsum
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-1wkb7z3"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-oxfbg2-MuiTypography-root"
        >
          Have fun!
        </h3>
        <img
          alt="card"
          src="function SvgImage3(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgImage3.name})
  );
}"
          style="max-width: 400px; flex: 1; margin-bottom: 24px; object-fit: cover;"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 css-1no9wdb-MuiTypography-root"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1ckkjjs-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Lorem Ipsum
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-1flv2gt"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 css-131agbk-MuiTypography-root"
    >
      Our Heroes
    </h2>
    <div
      class="MuiBox-root css-15hapvs"
    >
      <div
        class="MuiBox-root css-n6vyx9"
      >
        <div
          class="MuiBox-root css-1wkb7z3"
        >
          <img
            alt="card"
            src="anhelina_vrubleuskaya.jpg"
            style="max-width: 400px; max-height: 400px; flex: 1; margin-bottom: 24px; object-fit: cover; border-radius: 400px;"
          />
          <h4
            class="MuiTypography-root MuiTypography-h4 css-1psfiy1-MuiTypography-root"
          >
            Anhelina Vrubleuskaya
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1no9wdb-MuiTypography-root"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.
          </p>
        </div>
        <div
          class="MuiBox-root css-1wkb7z3"
        >
          <img
            alt="card"
            src="olga_polyanskaya.jpg"
            style="max-width: 400px; max-height: 400px; flex: 1; margin-bottom: 24px; object-fit: cover; border-radius: 400px;"
          />
          <h4
            class="MuiTypography-root MuiTypography-h4 css-1psfiy1-MuiTypography-root"
          >
            Olga Polyanskaya
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1no9wdb-MuiTypography-root"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.
          </p>
        </div>
        <div
          class="MuiBox-root css-1wkb7z3"
        >
          <img
            alt="card"
            src="konstantin_shumak.jpg"
            style="max-width: 400px; max-height: 400px; flex: 1; margin-bottom: 24px; object-fit: cover; border-radius: 400px;"
          />
          <h4
            class="MuiTypography-root MuiTypography-h4 css-1psfiy1-MuiTypography-root"
          >
            Konstantin Shumak
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1no9wdb-MuiTypography-root"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
